smoke/publish-local-registry.sh

```
@@ -59,11 +59,133 @@ do
      pnpm version $(cat $projectRoot/dist/fp-version) --no-git-tag-version &&
-     node $projectRoot/smoke/patch-fp-version.js package.json $(cat $projectRoot/dist/fp-version)
+     if [ -f "package.json" ]; then
+       node $projectRoot/smoke/patch-fp-version.js package.json $(cat $projectRoot/dist/fp-version)
+     else
+       echo "Warning: package.json not found in $smokeDir, skipping version patch"
+     fi &&
      cat .npmrc &&
      cat package.json &&
-     pnpm publish --registry=http://localhost:4873 --no-git-checks)
+     pnpm publish --registry=http://localhost:4873 --no-git-checks --tag smoke &&
+     npm dist-tag add $(node -e "console.log(require('./package.json').name)")@$(cat $projectRoot/dist/fp-version) latest --registry=http://localhost:4873)
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)